pkg/agent/informant.go

```diff
@@ -449,12 +449,11 @@ func (s *InformantServer) RegisterWithInformant(ctx context.Context, logger *zap
 							},
 						},
 					}
-					// TODO: fix context/logger stuff
 					s.runner.spawnBackgroundWorker(
-						context.Background(),
+						ctx,
 						disp.logger,
 						"dispatcher message handler",
-						func(context.Context, *zap.Logger) { disp.run() },
+						func(ctx context.Context, _ *zap.Logger) { disp.run(ctx) },
```

Comment:
```suggestion
						ignoreLogger(disp.run(ctx)),
```

I think this should do the trick (we we already have `ignoreLogger` around because this is just common enough to be worthwhile)